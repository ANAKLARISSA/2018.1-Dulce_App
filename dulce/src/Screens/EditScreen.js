import React from 'react';
import {View, TextInput, Text} from 'react-native';
import AGRButton from '../Components/AGRButton';


const styles = {
  container:{
    flex: 1,
    flexDirection: 'column',
    padding: 15,
    backgroundColor: '#FFF',
    paddingHorizontal: 20,
  },
  input:{
    height: 36,
    borderBottomWidth: 1,
    borderBottomColor: 'grey',
    marginBottom: 10,
    alignSelf: 'center',
    width: '80%',
    marginTop: 10,
  },
  button: {
    alignItems: 'center',
    backgroundColor: 'pink',
    padding: 10,
    color: 'blue'
  },
};

class EditScreen extends React.Component {
  constructor(props){
    super(props);
    this.state = {editable:false};
  }

  tornarVisivel(){
    this.setState({
      editable: !this.state.editable
    });
  }


  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input}
          placeholder='Nome'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Matricula'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Hospital'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Setor'
          editable = {this.state.editable}
        />
        <TextInput style={styles.input}
          placeholder='Editar senha'
          editable = {this.state.editable}
        />
        <AGRButton
          text ="Editar perfil"
          style={styles.button}
          onPress={() => {this.tornarVisivel()}}
        />

      </View>
    );
  }
}


export default EditScreen;
