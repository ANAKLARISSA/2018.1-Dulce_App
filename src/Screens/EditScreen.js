import React from 'react';
import {View} from 'react-native';
import ScreenHeader from '../Components/ScreenHeader';
import ValidationComponent from 'react-native-form-validator';
//import store from '../Reducers/store';
//import axios from 'axios';
import {Container} from 'native-base';
import t from 'tcomb-form-native';
import SignUpButton from '../Components/SignUpButton';

const Form = t.form.Form;

const styles = {
  container: {
    justifyContent: 'center',
    marginTop: 10,
    padding: 15,
    backgroundColor: '#FFF'
  },
  text: {
    fontSize: 15,
    marginBottom: 10,
    marginLeft: 10,
    color: 'black',
    fontFamily: 'Raleway'
  },
  button: {
    height: 45,
    paddingHorizontal: 50,
    backgroundColor: '#5f4b8b',
    alignSelf: 'center',
    alignItems: 'center',
    width: 145
  }
};


export default class EditScreen extends ValidationComponent {
  constructor(props) {
    super(props);
    this.Service = t.struct({
      nome: t.String,
      email: t.String,
      senha: t.String,
      confirmarSenha: t.String
    });
    this.state = {
      value: default_state
    };
    this.options = {
      fields: default_field_options,
      stylesheet: formStyles
    };
  }

  onChange(value) {
    this.setState({value});
  }
  render() {
    return (
<<<<<<< HEAD
      <Container>
        <ScreenHeader title='Editar conta' />
        <View style={styles.container}>
          <Form
            ref='form'
            options={this.options}
            value={this.state.value}
            type={this.Service}
            onChange={(v) => this.onChange(v)}
          />
        </View>
        <SignUpButton
          text = 'Editar'
          onPress = {() => {}}
          style = {styles.button}
        />
      </Container>
=======
      <View style={styles.container}>
      <ScreenHeader title='Editar Perfil' icon='arrow-back'/>
        <ScrollView>
          <SmallLogo/>
          <AGRInput style={styles.input} value = {this.state.name} editable = {false} nameLabel='Nome'/>
          {this.isFieldInError('name') && this.getErrorsInField('name').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.registration} editable = {false}
            placeholder='Matricula' nameLabel='Matrícula'/>
          <AGRInput value = {this.state.hospital} style={styles.input} editable = {this.state.editable}
            nameLabel='Hospital' onChangeText={(text) => this.setState({hospital: text})}/>
          {this.isFieldInError('hospital') && this.getErrorsInField('hospital').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} value = {this.state.sector} editable = {this.state.editable}
            nameLabel='setor' onChangeText={(text) => this.setState({sector: text})}/>
          {this.isFieldInError('sector') && this.getErrorsInField('sector').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <AGRInput style={styles.input} editable = {this.state.editable} nameLabel='Editar senha'
            secureTextEntry onChangeText={(text) => this.setState({password: text})}/>
          {this.isFieldInError('password') && this.getErrorsInField('password').map(errorMessage => <Text style={styles.error}>{errorMessage}</Text>) }
          <View style={styles.alinhar}>
            <AGRButton style={styles.button} text='Salvar' onPress={() => {this.save(); this._onPressButton();} }/>
          </View>
        </ScrollView>
      </View>
>>>>>>> 9930bc4... #386 Adicionando botões nas telas
    );
  }
}
const formStyles = {
  ...Form.stylesheet,
  error: {
    color: 'red',
    fontSize: 14,
    marginBottom: 3,
    fontWeight: '600'
  }
};

const default_field_options = {
  Nome: {
    error: 'Campo obrigatório'
  },
  confirmarSenha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  },
  Email: {
    error: 'Campo obrigatório'
  },
  senha: {
    secureTextEntry: true,
    error: 'Campo obrigatório'
  }
};

const default_state = {
  nome: '',
  email: '',
  senha: '',
  confirmarSenha: ''
};
