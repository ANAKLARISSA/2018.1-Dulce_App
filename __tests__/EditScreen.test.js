import 'react-native';
import React from 'react';
import EditScreen from '../src/Screens/EditScreen';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

require('bezier');

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer
  .create(<EditScreen />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

test('change name file ', () => {
  const wrapper = shallow(<EditScreen />);
  const nameField = wrapper.find('AGRInput').at(2);
  nameField.simulate('changeText','text');
  expect(wrapper.state('hospital')).toBe('text');
});

test('hospital field error' , () => {

  const wrapper = shallow(<EditScreen />);

  wrapper.setState({hospital: ""});

  wrapper.instance()._onPressButton();

  wrapper.setProps({});

  console.log(wrapper.debug());

  const nameFieldError = wrapper.find('Text').at(0);

  const fieldText = nameFieldError.dive().text();

  expect(fieldText).toBe('The field \"hospital\" is mandatory.');

});
