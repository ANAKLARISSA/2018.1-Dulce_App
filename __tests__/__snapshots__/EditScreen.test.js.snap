// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#FFF",
        "flex": 1,
        "flexDirection": "column",
        "padding": 15,
        "paddingHorizontal": 20,
      },
    ]
  }
  virtual={undefined}
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": undefined,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <ActivityIndicator
        animating={true}
        color="purple"
        hidesWhenStopped={true}
        size="large"
        style={
          Object {
            "height": 80,
          }
        }
        virtual={undefined}
      />
    </View>
  </RCTScrollView>
</View>
`;
